tutorial/README.md

```diff
@@ -0,0 +1,413 @@
+
+### Save Function - Writing to a Redshift table###
+
+spark-redshift allows you to write data back to Redshift. The source data is a `DataFrame` instance. We will need to assume that we have a source table available in the environment. This table can be sourced could be from a variety of sources such as a Hive table, CSV file, parquet file, delimited text file, etc. We will source a DataFrame from Redshift and write it back. This is for illustration purposes only. In practice the source would not be Redshift since in such a case it is easiest to perform a "CREATE TABLE AS" query on Redshift without going through Spark-SQL. 
+
+Listing 4 will register the `event` table as `myevent` table in the Spark environment.
+
+```scala
+val eventQuery = "select * from event"
+val eventDF = sqlContext.read
+             .format("com.databricks.spark.redshift")
+             .option("url",jdbcURL ) 
+             .option("tempdir", tempS3Dir)
+             .option("query", eventQuery) 
+             .load()   
+eventsDF.registerTempTable("myevent") 
+```
+Listing 4: Create a table myevent in the Spark environment
+
+Listing 5 demonstrates how to write the contents of `myevent` to a Redshift table `redshiftevent`
+
+```scala
+/*
+ *Create a new table redshiftevent after dropping any existing redshiftevent table
+ *and write event records with event id less than 1000
+ */
+sqlContext.sql("select * from myevent where eventid<=1000").withColumnRenamed("eventid", "id")
+             .write.format("com.databricks.spark.redshift")
+            .option("url", jdbcURL)
+            .option("tempdir", tempS3Dir)
+            .option("dbtable", "redshiftevent")
+            .mode(SaveMode.Overwrite) 
+            .save()
+/*
+ *Append to an existing table redshiftevent if it exists or create a new one if it does not
+ *exist and write event records with event id greater than 1000
+ */
+
+sqlContext.sql("select * from myevent where eventid>1000").withColumnRenamed("eventid", "id")
+          .write.format("com.databricks.spark.redshift")
+          .option("url", jdbcURL) 
+          .option("tempdir", tempS3Dir) 
+          .option("dbtable", "redshiftevent") 
+          .mode(SaveMode.Append)
+          .save()
+```
+Listing 5: Writing to Redshift tables
+
+spark-redshift automatically creates a Redshift table with the appropriate schema determined from the table/DataFrame being written. The default behavior is to create a new table and to throw an error message if a table with the same name already exists (corresponding to `SaveMode.ErrorIfExists`)
+
+There are two key points to note in Listing 5-
+
+1. Note the `.withColumnRenamed("eventid", "id")`. This feature if necessary if any of the source table columns names are key words in Redshift (ex. table) then spark-redshift will throw and error. This will be [fixed](https://github.com/databricks/spark-redshift/issues/80) in the next release 
```

Comment:
Note that we might very well release 0.5.1 by the time that we publish the post, in which case we'll want to update this.